components/ItemsPage/ItemsPage.tsx

```diff
@@ -111,21 +118,28 @@ const ItemsPage = (props: Props) => {
          )
          const finalInterface: Omit<MenuItemGroup, 'id'>[] = []
          for (let i = 0; i < quantity; i++) {
-            finalInterface.push({
-               menuItemId: theItem.id,
-               subTotal: price,
-               extraToppingIds: extraToppingIds,
-               extraSauceIds: extraSauceIds,
-               optionId: optionId[0],
-            })
+            optionId[0] !== undefined
```

Comment:
@brunorplima é aqui que aconteceu a mudança

Reply:
BTW, if this kind of problem happened and it couldn't be caught by the TS compiler, it means it's quite possible that we didn't define well the type(s) somewhere in the code, because this kind of problem is the kind we aim to avoid by using TypeScript. I'm not sure if it's worthwhile at this moment to look for where we didn't define well the typing but It's something to be on the lookout for in the next situations.